assets/js/data/cart/actions.ts

```diff
@@ -398,42 +390,40 @@ export function* removeItemFromCart(
  * @param {string} cartItemKey Cart item being updated.
  * @param {number} quantity    Specified (new) quantity.
  */
-export function* changeCartItemQuantity(
-	cartItemKey: string,
-	quantity: number
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- unclear how to represent multiple different yields as type
-): Generator< unknown, void, any > {
-	const cartItem = yield controls.resolveSelect(
-		CART_STORE_KEY,
-		'getCartItem',
-		cartItemKey
-	);
-	if ( cartItem?.quantity === quantity ) {
-		return;
-	}
-	yield itemIsPendingQuantity( cartItemKey );
-	try {
-		const { response } = yield apiFetchWithHeaders( {
-			path: '/wc/store/v1/cart/update-item',
-			method: 'POST',
-			data: {
-				key: cartItemKey,
-				quantity,
-			},
-			cache: 'no-store',
-		} );
-
-		yield receiveCart( response );
-	} catch ( error ) {
-		yield receiveError( error );
-
-		// If updated cart state was returned, also update that.
-		if ( error.data?.cart ) {
-			yield receiveCart( error.data.cart );
+export const changeCartItemQuantity =
+	(
+		cartItemKey: string,
+		quantity: number
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- unclear how to represent multiple different yields as type
+	) =>
+	async ( { dispatch, resolveSelect }: CartThunk ) => {
+		const cartItem = await resolveSelect.getCartItem( cartItemKey );
+		if ( cartItem?.quantity === quantity ) {
+			return;
 		}
-	}
-	yield itemIsPendingQuantity( cartItemKey, false );
-}
+		dispatch.itemIsPendingQuantity( cartItemKey );
+		try {
+			const { response } = await apiFetchWithHeaders( {
+				path: '/wc/store/v1/cart/update-item',
+				method: 'POST',
+				data: {
+					key: cartItemKey,
+					quantity,
+				},
+				cache: 'no-store',
+			} );
+
+			dispatch.receiveCart( response );
+		} catch ( error ) {
+			dispatch.receiveError( error );
+
+			// If updated cart state was returned, also update that.
+			if ( error.data?.cart ) {
+				dispatch.receiveCart( error.data.cart );
+			}
+		}
+		dispatch.itemIsPendingQuantity( cartItemKey, false );
```

Comment:
I've updated cases where it made sense in f237ab43bd1ebff99041d92b6aad3ca4c6b63a69. In 2e4a55254490a9919a0578274dcdbd985c820195 I added it to `removeItemFromCart` and `changeCartItemQuantity` because even if these actions error, we should still set the state to reflect that they're no longer removing or updating.